lms/djangoapps/learner_home/views.py

```diff
@@ -221,12 +242,29 @@ class InitializeView(RetrieveAPIView):  # pylint: disable=unused-argument
     """List of courses a user is enrolled in or entitled to"""
 
     def get(self, request, *args, **kwargs):  # pylint: disable=unused-argument
-        # Get user, determine if user needs to confirm email account
-        user = request.user
+        if request.GET.get('user'):
+            if not request.user.is_staff:
+                logger.info(f"[Learner Home] {request.user.username} attempted to masquerade but is not staff")
+                raise PermissionDenied()
+            try:
+                masquerade_username = request.GET.get('user')
+                masquerade_user = User.objects.get(username=masquerade_username)
+            except User.DoesNotExist:
```

Comment:
We should also support email as well. Maybe the params should be `masquerade_user=...`